flink-core/src/main/java/org/apache/flink/metrics/reporter/JMXReporter.java

```diff
@@ -265,4 +329,72 @@ public Object getValue() {
 			return gauge.getValue();
 		}
 	}
+
+	/**
+	 * JMX Server implementation that JMX clients can connect to.
+	 *
+	 * Heavily based on j256 simplejmx project
+	 *
+	 * https://github.com/j256/simplejmx/blob/master/src/main/java/com/j256/simplejmx/server/JmxServer.java
+	 */
+	private static class JMXServer {
+		private int port;
+		private Registry rmiRegistry;
+		private JMXConnectorServer connector;
+
+		public JMXServer(int port) {
+			this.port = port;
+		}
+
+		public void start() throws IOException {
+			startRmiRegistry();
+			startJmxService();
+		}
+
+		public void stop() throws IOException {
+			if (connector != null) {
+				try {
+					connector.stop();
+				} finally {
+					connector = null;
+				}
+			}
+			if (rmiRegistry != null) {
+				try {
+					UnicastRemoteObject.unexportObject(rmiRegistry, true);
+				} catch (NoSuchObjectException e) {
+					throw new IOException("Could not unexport our RMI registry", e);
+				} finally {
+					rmiRegistry = null;
+				}
+			}
+		}
+
+		private void startRmiRegistry() throws IOException {
+			if (rmiRegistry != null) {
+				return;
+			}
+			rmiRegistry = LocateRegistry.createRegistry(port);
+		}
+
+		private void startJmxService() throws IOException {
+			if (connector != null) {
+				return;
+			}
+			String serverHost = "localhost";
+			String registryHost = "";
+			String serviceUrl =
+				"service:jmx:rmi://" + serverHost + ":" + port + "/jndi/rmi://" + registryHost + ":" + port + "/jmxrmi";
```

Comment:
No, it does not allow connecting via remote. This also applies to the current version. 

I also remembered that this was not considered a required feature during implementation.